Add plugin: Inbox Organiser

jamiefdhurst opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/jamiefdhurst/obsidian-inbox-organiser

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
[this.vault.rename(file, `${path}/${file.name}`);](https://github.com/jamiefdhurst/obsidian-inbox-organiser/blob/f3b204ef702acb7ad4fa748d6261becb8e4748b0/src/inbox.ts#L27)
Prefer to use [FileManager.renameFile()](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/renameFile) to safely rename a file while updating links to that file.

[createFolderSelect(containerEl: HTMLElement): HTMLSelectElement {](https://github.com/jamiefdhurst/obsidian-inbox-organiser/blob/f3b204ef702acb7ad4fa748d6261becb8e4748b0/src/modal/index.ts#L129-L137)
I would recommend using the [PopoverSuggest](https://docs.obsidian.md/Reference/TypeScript+API/PopoverSuggest) component to enable type-ahead support for searching through the list of folders and to improve performance.

[return `${folder.name} (${parentNames.reverse().join(' > ')})`;](https://github.com/jamiefdhurst/obsidian-inbox-organiser/blob/f3b204ef702acb7ad4fa748d6261becb8e4748b0/src/modal/index.ts#L165)
Obsidian formats folders as `path/to/folder`, this format is going to feel out of place and potentially confusing.

jamiefdhurst commented:
Many thanks @Zachatoo - I've implemented an `AbstractInputSuggest` for the folders as suggested, which does work better and will be much more performant for users with higher folder counts - I agree. I've also switched out the vault method for the FileManager implementation. I believe that should cover everythig picked up in your review.

Please let me know if you need anything further.

jamiefdhurst commented:
Hi @joethei is anything else required here or is this good to add in? Please let me know as I can adjust more if needed.
